src/render/draw_line.js

```diff
@@ -38,91 +37,37 @@ export default function drawLine(painter: Painter, sourceCache: SourceCache, lay
         const prevProgram = painter.context.program.get();
         const program = painter.useProgram(programId, programConfiguration);
         const programChanged = firstTile || program.program !== prevProgram;
-        const tileRatioChanged = prevTileZoom !== tile.tileID.overscaledZ;
 
-        if (programChanged) {
-            programConfiguration.setUniforms(painter.context, program, layer.paint, {zoom: painter.transform.zoom});
-        }
-        drawLineTile(program, painter, tile, bucket, layer, coord, programConfiguration, programChanged, tileRatioChanged);
-        prevTileZoom = tile.tileID.overscaledZ;
+        drawLineTile(program, painter, tile, bucket, layer, coord, depthMode, colorMode, programConfiguration, programChanged);
         firstTile = false;
+        // once refactored so that bound texture state is managed, we'll also be able to remove this firstTile/programChanged logic
```

Comment:
Looks like the specific issue here is that `painter.lineAtlas.bind(context)` must currently be called _after_ `lineSDFUniformValues(...)`, because the latter produces side effects in the `lineAtlas` when it calls `lineAtlas.getDash(...)`.